server/server.go

```diff
@@ -288,10 +288,12 @@ func (s *Server) Modify(ms spb.GRIBI_ModifyServer) error {
 		for {
 			in, err := ms.Recv()
 			if err == io.EOF {
+				errCh <- nil
```

Comment:
I think this would immediately unblock the following `err := <-errCh` and then `s.deleteClient(cid)`, which might not give the server a chance to finish sending out all the results, right?

Reply:
Yes theoretically the server should finish up, but currently it doesn't do that. This line addresses the problem that a bunch of compliance tests would deadlock because the clients are waiting for the server to close which never happened.

This seems like the most straightforward solution for now. If you want to improve on it, feel free to send a follow-up PR.